src/frontend/src/handler/drop_table.rs

```diff
@@ -57,10 +63,127 @@ pub async fn handle_drop_table(
         if table.table_type() != TableType::Table {
             return Err(table.bad_drop_error());
         }
-
-        (table.associated_source_id(), table.id())
+        (table.associated_source_id(), table.id(), table.engine)
     };
 
+    match engine {
+        Engine::Iceberg => {
+            let either = if let Ok(source) = session
+                .env()
+                .catalog_reader()
+                .read_guard()
+                .get_source_by_name(
+                    db_name,
+                    schema_path,
+                    &(ICEBERG_SOURCE_PREFIX.to_string() + &table_name),
+                )
+                .map(|(source, _)| source.clone())
+            {
+                let config = ConnectorProperties::extract(source.with_properties.clone(), false)?;
+                if let ConnectorProperties::Iceberg(iceberg_properties) = config {
+                    Some(Either::Left(iceberg_properties))
+                } else {
+                    unreachable!("must be iceberg source");
+                }
+            } else if let Ok(sink) = session
+                .env()
+                .catalog_reader()
+                .read_guard()
+                .get_sink_by_name(
+                    db_name,
+                    schema_path,
+                    &(ICEBERG_SINK_PREFIX.to_string() + &table_name),
+                )
+                .map(|(sink, _)| sink.clone())
+            {
+                // If iceberg source does not exist, use iceberg sink to load iceberg table
+                let iceberg_config = IcebergConfig::from_btreemap(sink.properties.clone())?;
+                Some(Either::Right(iceberg_config))
+            } else {
+                None
+            };
+
+            // TODO(iceberg): make iceberg engine table drop ddl atomic
```

Comment:
I think the current approach is good and no need for the `TODO`. It can ensure the table, source, and sink to be dropped as long as one of them exists.